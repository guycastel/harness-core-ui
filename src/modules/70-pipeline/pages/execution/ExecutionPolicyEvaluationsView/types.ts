export enum EvaluationStatus {
  ERROR = 'error',
  PASS = 'pass',
  WARNING = 'warning'
}

export interface IacmFormattedData {
  policySetId?: string
  policySetName?: string
  status?: string
  identifier?: string
  created?: string | number
  accountId?: string
  orgId?: string
  projectId?: string
  description?: string
  policyMetadata: {
    id?: string | number
    policyId?: string
    policyName?: string
    severity?: string
    denyMessages?: string[]
    status?: string
    identifier?: string | number
    accountId?: string
    orgId?: string
    projectId?: string
    created?: string | number
    updated?: string | number
    error?: string
  }[]
}
