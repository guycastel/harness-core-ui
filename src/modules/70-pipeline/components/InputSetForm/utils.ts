/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { SaveToGitFormInterface } from '@modules/10-common/components/SaveToGitForm/SaveToGitForm'
import type { EntityGitDetails } from 'services/pipeline-ng'

export interface GetUpdatedGitDetailsReturnType extends EntityGitDetails {
  lastObjectId?: string
  lastCommitId?: string
  baseBranch?: string
}

export const getUpdatedGitDetails = (
  isEdit: boolean,
  gitDetails: SaveToGitFormInterface | undefined,
  lastObjectId: string,
  initialGitDetails: EntityGitDetails,
  conflictCommitId?: string
): GetUpdatedGitDetailsReturnType => {
  let updatedGitDetails: GetUpdatedGitDetailsReturnType = {}
  if (gitDetails) {
    updatedGitDetails = { ...gitDetails }
    if (isEdit) {
      updatedGitDetails['lastObjectId'] = lastObjectId
      updatedGitDetails['lastCommitId'] = conflictCommitId || initialGitDetails.commitId
    }
    if (gitDetails.isNewBranch) {
      updatedGitDetails['baseBranch'] = initialGitDetails.branch
    }
  }
  return updatedGitDetails
}
