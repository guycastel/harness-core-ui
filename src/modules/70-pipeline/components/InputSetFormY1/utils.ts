/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { cloneDeep, defaultTo, forOwn, pick } from 'lodash-es'
import { CompletionItemKind } from 'vscode-languageserver-types'
import { GitDetails } from '@harnessio/react-pipeline-service-client'
import { CompletionItemInterface } from '@modules/10-common/interfaces/YAMLBuilderProps'
import { EntityGitDetails } from 'services/pipeline-ng'
import { StoreMetadata } from '@modules/10-common/constants/GitSyncTypes'
import {
  FromikInputSetY1,
  InputSetKVPairs,
  InputSetMetadataY1,
  InputSetY1,
  InputSetsY1,
  SelectedItemsType
} from './types'
import { SelectedInputSetListValue } from '../InputSetSelector/SelectedInputSetList'

export const inputSetIdsToSelectedItems = (
  inputSetIds: InputSetsY1,
  allInputSetItems: SelectedItemsType[]
): SelectedItemsType[] => {
  return allInputSetItems.filter(item => inputSetIds.includes(item.value as string))
}

export const getInputSetFromYaml = (
  values: InputSetY1,
  options: { escapeEmpty: boolean } = { escapeEmpty: false }
): InputSetY1 => {
  const ret = cloneDeep(values)

  if (ret.spec) {
    const spec = ret.spec
    if (options.escapeEmpty) {
      forOwn(spec, (value: unknown, key: unknown) => {
        if (key === 'input_sets') return

        if (typeof key === 'string' && typeof value === 'string') {
          spec[key as string] = value === '' ? '""' : value
        }
      })
    }
  }

  return ret
}

export const replaceEmptyWithNull = (values: FromikInputSetY1): { changed: boolean; values: FromikInputSetY1 } => {
  let changed = false
  const retValues = cloneDeep(values)

  forOwn(values.spec, (value: unknown, key: unknown) => {
    if (key === 'input_sets') return

    if (typeof key === 'string') {
      if (retValues.spec) {
        if (value === '' || value === null) {
          changed = true
          retValues.spec[key as string] = null
        } else {
          retValues.spec[key as string] = value
        }
      }
    }
  })

  return { changed, values: retValues }
}

export const addRemoveKeysFromInputSet = (
  inputSet: FromikInputSetY1,
  visibleKeys: string[],
  mergedInputSetValues: InputSetKVPairs
): FromikInputSetY1 => {
  const newInputSet = cloneDeep(inputSet)
  const spec = newInputSet.spec ?? {}

  // add
  visibleKeys.forEach(key => {
    // apply merged or null
    if (typeof spec?.[key] === 'undefined') spec[key] = mergedInputSetValues[key] ?? null
  })

  // remove
  Object.keys(spec).forEach(key => {
    if (!visibleKeys.includes(key) && key !== 'input_sets') {
      delete spec[key]
    }
  })

  newInputSet.spec = spec

  return newInputSet
}

export const getDefaultInputSet = (): InputSetY1 => {
  return {
    version: 1,
    kind: 'input-set'
  }
}
export const formikToYaml = (values: FromikInputSetY1, options?: { escapeEmpty?: boolean }): InputSetY1 => {
  const ret = {
    version: 1,
    kind: 'input-set',
    spec: values.spec
  }

  if (ret.spec) {
    const spec = ret.spec
    if (spec && options?.escapeEmpty) {
      forOwn(ret.spec, (value: unknown, key: unknown) => {
        if (typeof key === 'string' && typeof value === 'string') {
          spec[key as string] = value === '""' ? '' : value
        }
      })
    }
  }

  return ret
}

export const formikToMetadata = (values: FromikInputSetY1): InputSetMetadataY1 => {
  return pick(values, 'name', 'identifier', 'description', 'tags')
}

export const formikToGitMetadata = (values: FromikInputSetY1): StoreMetadata => {
  return {
    branch: values.branch,
    connectorRef: values.connectorRef as string,
    repoName: values.repo,
    filePath: values.filePath,
    storeType: values.storeType
  }
}

export const constructInputSetYamlObject = (item: SelectedInputSetListValue): CompletionItemInterface => ({
  label: defaultTo(item.label, ''),
  insertText: defaultTo(item.value as string, ''),
  kind: CompletionItemKind.Field
})

export const gitDetailsV1toV0 = (gitDetails?: GitDetails): EntityGitDetails | undefined => {
  if (gitDetails) {
    // TODO check missing properties
    return {
      branch: gitDetails.branch_name,
      commitId: gitDetails.commit_id,
      filePath: gitDetails.file_path,
      fileUrl: gitDetails.file_url,
      // isHarnessCodeRepo?: ?
      objectId: gitDetails.object_id,
      // parentEntityConnectorRef: ?,
      // parentEntityRepoName: ?
      // repoIdentifier: ?
      repoName: gitDetails.repo_name,
      repoUrl: gitDetails.repo_url
      // rootFolder: ?
    }
  }
}

export const gitDetailsV0toV1 = (gitDetails: EntityGitDetails): GitDetails | undefined => {
  if (gitDetails) {
    // TODO check missing properties
    return {
      branch_name: gitDetails.branch,
      commit_id: gitDetails.commitId,
      file_path: gitDetails.filePath,
      file_url: gitDetails.fileUrl,
      // isHarnessCodeRepo?: ?
      object_id: gitDetails.objectId,
      // parentEntityConnectorRef: ?,
      // parentEntityRepoName: ?
      // repoIdentifier: ?
      repo_name: gitDetails.repoName,
      repo_url: gitDetails.repoUrl
      // rootFolder: ?
    }
  }
}
