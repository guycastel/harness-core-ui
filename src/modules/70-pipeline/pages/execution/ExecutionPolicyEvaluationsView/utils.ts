import { defaultTo, forEach, includes, map } from 'lodash-es'
import { GovernanceMetadata } from 'services/cd-ng'
import { Evaluation } from 'services/pm'
import { EvaluationStatus, IacmFormattedData } from './types'

const defaultEmpty = (value: string | number | undefined): string => defaultTo(value, '') as string

const formatIacmGovernanceData = (
  data?: Evaluation[]
): {
  iacmData: IacmFormattedData[]
  iacmStatus: EvaluationStatus
} => {
  const iacmData: IacmFormattedData[] = []
  let status: EvaluationStatus = EvaluationStatus.PASS
  const statuses: EvaluationStatus[] = []
  forEach(data, ({ id, details }) =>
    forEach(details, policySet => {
      statuses.push(defaultTo(policySet.status as EvaluationStatus, EvaluationStatus.PASS))
      iacmData.push({
        policySetId: defaultEmpty(policySet.identifier),
        policySetName: defaultEmpty(policySet.name),
        status: defaultEmpty(policySet.status),
        identifier: defaultEmpty(policySet.identifier),
        created: defaultEmpty(policySet.created),
        accountId: defaultEmpty(policySet.account_id),
        orgId: defaultEmpty(policySet.org_id),
        projectId: defaultEmpty(policySet.project_id),
        policyMetadata: map(policySet.details, policy => ({
          id: id,
          policyId: defaultEmpty(policy?.policy?.identifier),
          policyName: defaultEmpty(policy?.policy?.name),
          severity: defaultEmpty(policySet.status),
          denyMessages: policy?.deny_messages,
          status: defaultEmpty(policySet.status),
          identifier: defaultEmpty(policy?.policy?.identifier),
          accountId: defaultEmpty(policySet.account_id),
          orgId: defaultEmpty(policySet.org_id),
          projectId: defaultEmpty(policySet.project_id),
          created: defaultEmpty(policySet.created),
          updated: defaultEmpty(policySet.updated)
        }))
      })
    })
  )

  if (includes(statuses, EvaluationStatus.ERROR)) {
    status = EvaluationStatus.ERROR
  } else if (includes(statuses, EvaluationStatus.WARNING)) {
    status = EvaluationStatus.WARNING
  }
  return { iacmData, iacmStatus: status }
}

export const formatMetaData = (
  isLoading: boolean,
  IACM_ENABLED: boolean,
  governanceMetadata?: GovernanceMetadata,
  data?: Evaluation[]
): GovernanceMetadata => {
  const details: IacmFormattedData[] = []
  let govData = {
    status: EvaluationStatus.PASS,
    id: null,
    details
  }

  if (governanceMetadata) {
    const { details: opaDetails, status: opaStatus, id } = governanceMetadata
    govData = {
      details: opaDetails,
      status: opaStatus,
      id
    }
  }

  const { iacmData, iacmStatus } = formatIacmGovernanceData(data)
  if (IACM_ENABLED && iacmData && !isLoading) {
    govData.details = [...defaultTo(govData.details, []), ...iacmData]
  }

  if (IACM_ENABLED && iacmStatus !== govData.status) {
    if (govData.status === EvaluationStatus.ERROR || iacmStatus === EvaluationStatus.ERROR) {
      govData.status = EvaluationStatus.ERROR
    } else if (govData.status === EvaluationStatus.WARNING || iacmStatus === EvaluationStatus.WARNING) {
      govData.status = EvaluationStatus.WARNING
    } else {
      govData.status = EvaluationStatus.PASS
    }
  }

  return govData
}
