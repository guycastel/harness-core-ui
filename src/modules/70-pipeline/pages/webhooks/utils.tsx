/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { v4 as nameSpace, v5 as uuid } from 'uuid'
import { IconProps, SelectOption } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { RBACError } from '@rbac/utils/useRBACError/useRBACError'
import { StringsMap } from 'stringTypes'
import { StringKeys } from 'framework/strings'

export interface AddWebhookModalData {
  name: string
  identifier: string
  connectorRef: string
  repo: string
  folderPaths: { id: string; value: string }[]
}

export interface NewWebhookModalProps {
  isEdit: boolean
  initialData: AddWebhookModalData
  entityScope?: Scope
  closeModal: () => void
}
export function processFolderPaths(folderPaths: string[]): { id: string; value: string }[] {
  return folderPaths.map(path => ({ id: uuid('', nameSpace()), value: path }))
}

export const initialWebhookModalData = {
  name: '',
  identifier: '',
  connectorRef: '',
  repo: '',
  folderPaths: processFolderPaths([''])
}

export enum STATUS {
  'loading',
  'error',
  'ok'
}

export interface Error {
  message: RBACError
}

export enum WebhookTabIds {
  ListTab = 'ListTab',
  EventsTab = 'EventsTab'
}

export type WebhookEventStatus = 'FAILED' | 'PROCESSING' | 'QUEUED' | 'SKIPPED' | 'SUCCESSFUL'

export const stringsMap: Readonly<Record<WebhookEventStatus, StringKeys>> = {
  FAILED: 'pipeline.executionFilters.labels.Failed',
  PROCESSING: 'pipeline.eventStatus.processing',
  QUEUED: 'pipeline.eventStatus.queued',
  SKIPPED: 'pipeline.testsReports.skipped',
  SUCCESSFUL: 'pipeline.executionFilters.labels.Success'
}

export function getStatusList(getString: (key: keyof StringsMap) => string): SelectOption[] {
  return Object.keys(stringsMap).map(status => {
    return {
      label: getString(stringsMap[status as WebhookEventStatus]),
      value: status
    }
  })
}

export const iconMap: Record<WebhookEventStatus, IconProps> = {
  SUCCESSFUL: { name: 'tick-circle', size: 10 },
  FAILED: { name: 'warning-sign', size: 9 },
  PROCESSING: { name: 'loading', size: 10 },
  QUEUED: { name: 'queued', size: 10 },
  SKIPPED: { name: 'skipped', size: 8 }
}
