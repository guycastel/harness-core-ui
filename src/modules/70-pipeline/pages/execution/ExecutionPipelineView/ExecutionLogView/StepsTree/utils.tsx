/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo } from 'lodash-es'
import {
  ExecutionPipelineGroupInfo,
  ExecutionPipelineItem,
  ExecutionPipelineNode
} from '@pipeline/components/ExecutionStageDiagram/ExecutionPipelineModel'
import { ExecutionNode, InterruptEffectDTO } from 'services/pipeline-ng'
import { getExecutionNodeName } from '@pipeline/utils/execUtils'

export function getRetryInterrupts(step: ExecutionPipelineNode<ExecutionNode>): InterruptEffectDTO[] {
  return defaultTo(step?.item?.data?.interruptHistories, defaultTo(step?.group?.data?.interruptHistories, [])).filter(
    row => row.interruptType === 'RETRY'
  )
}

export function getStepDisplayName(
  step: ExecutionPipelineItem<ExecutionNode> | ExecutionPipelineGroupInfo<ExecutionNode>
): string {
  return getExecutionNodeName(step.data) || step.name
}
