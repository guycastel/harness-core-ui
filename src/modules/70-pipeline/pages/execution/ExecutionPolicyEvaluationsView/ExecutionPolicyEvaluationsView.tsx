/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useMemo, useEffect } from 'react'
import { useParams } from 'react-router-dom'
import { Container, useToaster } from '@harness/uicore'
import type { ExecutionPathProps, PipelineType } from '@common/interfaces/RouteInterfaces'
import { useExecutionContext } from '@pipeline/context/ExecutionContext'
import { PolicyManagementEvaluationView } from '@governance/PolicyManagementEvaluationView'
import { FeatureFlag } from '@modules/10-common/featureFlags'
import { useFeatureFlag } from '@modules/10-common/hooks/useFeatureFlag'
import { Evaluation, useGetEvaluationList } from 'services/pm'
import { formatMetaData } from './utils'

export default function ExecutionPolicyEvaluationsView(): React.ReactElement | null {
  const { showError } = useToaster()
  const IACM_ENABLED = useFeatureFlag(FeatureFlag.IACM_ENABLED)
  const { accountId, orgIdentifier, projectIdentifier, executionIdentifier, module } =
    useParams<PipelineType<ExecutionPathProps>>()
  const context = useExecutionContext()
  const governanceMetadata = context?.pipelineExecutionDetail?.pipelineExecutionSummary?.governanceMetadata
  const queryParams = {
    accountIdentifier: accountId,
    orgIdentifier,
    projectIdentifier,
    includeChildScopes: true,
    module: 'iacm',
    entity: `iacm-${executionIdentifier}`,
    lazy: !IACM_ENABLED
  }
  const { data, loading, error } = useGetEvaluationList({ queryParams })

  useEffect(() => {
    if (error) {
      showError(error)
    }
  }, [error])

  const metaData = useMemo(() => {
    if (!data && !governanceMetadata) {
      return null
    }
    return formatMetaData(loading, IACM_ENABLED, governanceMetadata, data as Evaluation[])
  }, [governanceMetadata, data, IACM_ENABLED, loading])

  if (!metaData) {
    return null
  }

  return (
    <Container width="100%" height="100%">
      <PolicyManagementEvaluationView metadata={metaData} accountId={accountId} module={module} />
    </Container>
  )
}
