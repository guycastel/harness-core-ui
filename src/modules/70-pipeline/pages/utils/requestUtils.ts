/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { COMMON_DEFAULT_PAGE_SIZE } from '@common/constants/Pagination'
import { CommonPaginationQueryParams } from '@common/hooks/useDefaultPaginationProps'
import { useFeatureFlags } from '@common/hooks/useFeatureFlag'
import { UseQueryParamsOptions, useQueryParamsOptions } from '@common/hooks/useQueryParams'
import { WebhookEventStatus } from '../webhooks/utils'

export interface EventsDateFilter {
  startTime: number
  endTime: number
}

export type WebhookEventsQueryParams = {
  dateFilter?: EventsDateFilter
  webhookIdentifier?: string
  eventId?: string
  eventStatus?: WebhookEventStatus[]
} & CommonPaginationQueryParams

export type WebhookEventsQueryParamsWithDefaults = RequiredPick<
  WebhookEventsQueryParams,
  'page' | 'size' | 'dateFilter'
>

export const WEBHOOK_EVENTS_PAGE_INDEX = 0
export const WEBHOOK_EVENTS_PAGE_SIZE = 20

export const useWebhookEventsQueryParamOptions = (): UseQueryParamsOptions<WebhookEventsQueryParamsWithDefaults> => {
  const { PL_NEW_PAGE_SIZE } = useFeatureFlags()

  const start = new Date()
  start.setDate(start.getDate() - 7)
  start.setHours(0, 0, 0, 0)
  const end = new Date()
  end.setHours(23, 59, 59, 999)

  return useQueryParamsOptions({
    page: WEBHOOK_EVENTS_PAGE_INDEX,
    size: PL_NEW_PAGE_SIZE ? COMMON_DEFAULT_PAGE_SIZE : WEBHOOK_EVENTS_PAGE_SIZE,
    dateFilter: {
      startTime: start.getTime(),
      endTime: end.getTime()
    }
  })
}
