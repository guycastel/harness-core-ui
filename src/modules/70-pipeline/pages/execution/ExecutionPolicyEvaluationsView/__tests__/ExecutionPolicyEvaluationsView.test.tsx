/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import mockImport from 'framework/utils/mockImport'
import * as pm from 'services/pm'
import ExecutionPolicyEvaluationsView from '../ExecutionPolicyEvaluationsView'

jest.mock('services/pm', () => ({
  useGetEvaluationList: jest.fn(() => ({
    data: null,
    loading: false,
    refetch: jest.fn().mockReturnValue({}),
    error: null
  }))
}))

describe('<ExecutionPolicyEvaluationsView /> tests', () => {
  test('Should render empty when governanceMetadata is null', () => {
    mockImport('@pipeline/context/ExecutionContext', {
      useExecutionContext: jest.fn()
    })
    mockImport('@governance/PolicyManagementEvaluationView', {
      PolicyManagementEvaluationView: () => <div /> // eslint-disable-line react/display-name
    })

    const { container } = render(
      <TestWrapper>
        <ExecutionPolicyEvaluationsView />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should render empty when governanceMetadata is not null', () => {
    mockImport('@pipeline/context/ExecutionContext', {
      useExecutionContext: () => {
        return {
          pipelineExecutionDetail: {
            pipelineExecutionSummary: {
              governanceMetadata: {}
            }
          }
        }
      }
    })
    mockImport('@governance/PolicyManagementEvaluationView', {
      PolicyManagementEvaluationView: () => <div /> // eslint-disable-line react/display-name
    })

    const { container } = render(
      <TestWrapper>
        <ExecutionPolicyEvaluationsView />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should render empty when iacm data status of error', () => {
    jest.spyOn(pm, 'useGetEvaluationList').mockImplementation((): any => ({
      data: [
        {
          id: 'mock_id',
          details: [
            {
              status: 'error',
              identifier: 'mock_identifier',
              name: 'mock_name',
              created: 'mock_date',
              account_id: 'mock_date',
              org_id: 'mock_date',
              project_id: 'mock_date',
              details: [
                {
                  status: 'error',
                  deny_messages: ['mock_deny_message'],
                  account_id: 'mock_account_id',
                  org_id: 'mock_org_id',
                  project_id: 'mock_project_id',
                  created: 'mock_date',
                  updated: 'mock_date',
                  policy: {
                    identifier: 'mock_policy_id',
                    name: 'mock_policy_name'
                  }
                }
              ]
            }
          ]
        }
      ],
      loading: false,
      error: null
    }))
    mockImport('@pipeline/context/ExecutionContext', {
      useExecutionContext: () => {
        return {
          pipelineExecutionDetail: {
            pipelineExecutionSummary: {
              governanceMetadata: {}
            }
          }
        }
      }
    })
    mockImport('@governance/PolicyManagementEvaluationView', {
      PolicyManagementEvaluationView: () => <div /> // eslint-disable-line react/display-name
    })

    const { container } = render(
      <TestWrapper defaultFeatureFlagValues={{ IACM_ENABLED: true }}>
        <ExecutionPolicyEvaluationsView />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should render empty when iacm data status of warning', () => {
    jest.spyOn(pm, 'useGetEvaluationList').mockImplementation((): any => ({
      data: [
        {
          id: 'mock_id',
          details: [
            {
              status: 'warning',
              details: [
                {
                  policy: {}
                }
              ]
            }
          ]
        }
      ],
      loading: false,
      error: null
    }))
    mockImport('@pipeline/context/ExecutionContext', {
      useExecutionContext: () => {
        return {
          pipelineExecutionDetail: {
            pipelineExecutionSummary: {
              governanceMetadata: {}
            }
          }
        }
      }
    })
    mockImport('@governance/PolicyManagementEvaluationView', {
      PolicyManagementEvaluationView: () => <div /> // eslint-disable-line react/display-name
    })

    const { container } = render(
      <TestWrapper defaultFeatureFlagValues={{ IACM_ENABLED: true }}>
        <ExecutionPolicyEvaluationsView />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
