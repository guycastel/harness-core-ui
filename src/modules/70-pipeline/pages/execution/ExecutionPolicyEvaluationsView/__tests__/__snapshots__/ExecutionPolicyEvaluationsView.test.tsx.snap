// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExecutionPolicyEvaluationsView /> tests Should render empty when governanceMetadata is not null 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    style="width: 100%; height: 100%;"
  >
    <div />
  </div>
</div>
`;

exports[`<ExecutionPolicyEvaluationsView /> tests Should render empty when governanceMetadata is null 1`] = `<div />`;

exports[`<ExecutionPolicyEvaluationsView /> tests Should render empty when iacm data status of error 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    style="width: 100%; height: 100%;"
  >
    <div />
  </div>
</div>
`;

exports[`<ExecutionPolicyEvaluationsView /> tests Should render empty when iacm data status of warning 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    style="width: 100%; height: 100%;"
  >
    <div />
  </div>
</div>
`;
